pkg/updatestatus/controlplaneinformer_test.go

```diff
@@ -1244,10 +1317,108 @@ func Test_assessClusterVersion(t *testing.T) {
 					Conditions: []configv1.ClusterOperatorStatusCondition{tc.cvProgressing},
 				},
 			}
-			actual := assessClusterVersion(cv, now)
-			if diff := cmp.Diff(tc.expected, actual, cmpopts.IgnoreFields(metav1.Condition{}, "LastTransitionTime")); diff != "" {
+			actualCvStatusInsight, healthInsights := assessClusterVersion(cv, now)
+			if diff := cmp.Diff(tc.expected, actualCvStatusInsight, cmpopts.IgnoreFields(metav1.Condition{}, "LastTransitionTime")); diff != "" {
 				t.Errorf("CV Status Insight differs from expected:\n%s", diff)
 			}
+
+			if diff := cmp.Diff([]*HealthInsight(nil), healthInsights); diff != "" {
+				t.Errorf("Unexpected health insights:\n%s", diff)
+			}
+		})
+	}
+}
+
+func Test_assessClusterVersion_testingHealthInsight(t *testing.T) {
+	now := metav1.Now()
+	var minutesAgo [30]metav1.Time
+	for i := range minutesAgo {
+		minutesAgo[i] = metav1.NewTime(now.Add(-time.Duration(i) * time.Minute))
+	}
+
+	cvReference := ResourceRef{
+		Resource: "clusterversions",
+		Group:    "config.openshift.io",
+		Name:     "version",
+	}
+
+	testCases := []struct {
+		name string
+
+		cvAnnotations map[string]string
+
+		expected []*HealthInsight
+	}{
+		{
+			name:          "no annotations -> no insight",
+			cvAnnotations: nil,
+			expected:      nil,
+		},
+		{
+			name: "unrelated annotations -> no insight",
+			cvAnnotations: map[string]string{
+				"foo": "bar",
+			},
+			expected: nil,
+		},
+		{
+			name: "usc.openshift.io/force-health-insight=true -> health insight",
+			cvAnnotations: map[string]string{
+				"usc.openshift.io/force-health-insight": "true",
+			},
+			expected: []*HealthInsight{
+				{
+					StartedAt: now,
+					Scope: InsightScope{
+						Type:      ControlPlaneScope,
+						Resources: []ResourceRef{cvReference},
+					},
+					Impact: InsightImpact{
+						Level:       InfoImpactLevel,
+						Type:        NoneImpactType,
+						Summary:     "Forced health insight for ClusterVersion version",
+						Description: "The resource has a `usc.openshift.io/force-health-insight` annotation which forces USC to generate this health insight for testing purposes.",
+					},
+					Remediation: InsightRemediation{
+						Reference: "https://issues.redhat.com/browse/OTA-1418",
+					},
+				},
+			},
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			cv := &configv1.ClusterVersion{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:        "version",
+					Annotations: tc.cvAnnotations,
+				},
+				Status: configv1.ClusterVersionStatus{
```

Comment:
Do we need the status here?
It works for me without it.

Reply:
I guess we can get rid of it. `assessClusterVersion` is supposed to read it but apparently it survives and returns _some_ CV status insight even when status is empty.

Reply:
Addressed in https://github.com/openshift/cluster-version-operator/compare/dab7e4cf86b6fb08b7b08900cf11232912bb3545..c0e26a4186e4e248a5a0da8b505bb2564e75510a